### PR TITLE
Add unit types to CombatInfoComponent

### LINKED ISSUE #491: Add the attacking and defending army to the CombatInfoComponent
Right now we have to check the GameLog to see what units have been chosen for attacking. It would be nice to see that directly in the CombatInfoComponent.

### PR DESCRIPTION
**Pull request by gereon77**
Closes #491 

Preview:

image

Migration for the game log is missing. If ok, for past battles I would set armyUnits to an empty array.

### CODE REVIEW

**Comment by Longwelwind** [comment]
LGTM. Can you do the migration in this PR too? Or do you prefer that I do it?

**Comment by gereon77** [comment]
> 
> 
> LGTM. Can you do the migration in this PR too? Or do you prefer that I do it?

I try it this evening.

**Comment by gereon77** [comment]
I added the migration. Hope it's ok.

File: agot-bg-game-server/src/server/serializedGameMigrations.ts [hunk 1]
```diff
@@ -38,6 +38,23 @@ const serializedGameMigrations: {version: string; migrate: (serializeGamed: any)
...
+        version: "3",
+        migrate: (serializedGame: any) => {
+            // Migration for #506
+            if (serializedGame.childGameState.type == "ingame") {
+                const ingame = serializedGame.childGameState;
+
+                ingame.gameLogManager.logs
+                    .filter((log: any) => log.data.type == "combat-result")
+                    .forEach((log: any) => {
+                        log.data.armyUnits = []
```
Comment (Longwelwind) [inline agot-bg-game-server/src/server/serializedGameMigrations.ts:54]: It's:

```
.forEach((log: any) => {
  log.data.stats.forEach(s => s.armyUnits = []);
});
```

Comment (gereon77) [reply agot-bg-game-server/src/server/serializedGameMigrations.ts]: Ah yea, thx. :) I changed it.

**Comment by Longwelwind** [comment]
LGTM.

Rebase and I'll merge.

**Pull request merged**